pytorch_lightning/accelerators/accelerator.py

```diff
@@ -338,12 +338,21 @@ def lightning_module_state_dict(self) -> Dict[str, Union[Any, Tensor]]:
         return self.training_type_plugin.lightning_module_state_dict()
 
     def barrier(self, name: Optional[str] = None) -> None:
+        """
```

Comment:
add a deprecation warning here too so it shows up at runtime. 
then we can add a deprecation test for it here: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/deprecated_api/test_remove_1-6.py

it's not foolproof, since someone with a custom accelerator could override the whole method w/o calling super().barrier() but as far as i can tell this is the best we can do now

Reply:
all these three are class function, and there is no reference in entire code base anymore. I don't know how can we add deprecation test. Any idea?

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/blob/8fcdcb598bbf0fbe6cda92415d1d69eb1e2d0aa0/tests/accelerators/test_cpu.py#L42-L47

something like this?

where you directly call `accelerator.barrier()` and confirm the deprecation message appears

Reply:
Added!